lib/pacemaker/pcmk_sched_location.c

```diff
@@ -262,6 +323,13 @@ unpack_rsc_location(xmlNode *xml_obj, pcmk_resource_t *rsc,
             return;
         }
 
+        rc = pcmk_parse_score(score, &score_i, 0);
+        if (rc != pcmk_rc_ok) { // Not possible with schema validation enabled
+            pcmk__config_err("Ignoring location constraint %s "
+                             "because '%s' is not a valid score", id, score);
```

Comment:
Doesn't look like we're `NULL`-checking `id`; that's true for several existing log messages in this function too.

Reply:
Actually it was done: `pcmk__unpack_constraints()` ignores any constraint without an ID, so all of the constraint-specific unpacking can assume non-NULL